File: ios/Plugin/SerialPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(SerialPlugin, "Serial",
           CAP_PLUGIN_METHOD(closeConnection, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(openConnection, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(read, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(registerReadCallback, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(registerReadRawCallback, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestSerialPermissions, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(unregisterReadCallback, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(unregisterReadRawCallback, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(write, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(writeHexadecimal, CAPPluginReturnPromise);
)

File: ios/Plugin/SerialPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SerialPlugin)
public class SerialPlugin: CAPPlugin {
    private let implementation = Serial()

   @objc func closeConnection(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

   @objc func openConnection(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

  @objc func read(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

 @objc func registerReadCallback(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

 @objc func requestSerialPermissions(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

  @objc func unregisterReadCallback(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

  @objc func write(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

  @objc func writeHexadecimal(_ call: CAPPluginCall) {
        call.unimplemented("Not implemented on iOS.")
  }

}
